common/wallet_tx.c

```diff
@@ -39,6 +48,59 @@ struct command_result *param_wtx(struct command *cmd,
         return NULL;
 }
 
+struct command_result *param_wtx_utxos(struct command *cmd,
+				 const char *name,
+				 const char *buffer,
+				 const jsmntok_t *tok,
+				 struct wallet_tx *wtx)
+{
+	size_t i;
+	const jsmntok_t *curr;
+	struct bitcoin_txid **txids;
+	unsigned int **outnums;
+	char **outpoint = tal_arr(tmpctx, char*, 0);
+
+	txids = tal_arr(cmd, struct bitcoin_txid*, 0);
+	outnums = tal_arr(cmd, unsigned int*, 0);
+	json_for_each_arr(i, curr, tok) {
+		outpoint = tal_strsplit(cmd, json_strdup(cmd, buffer, curr), ":",
+				STR_NO_EMPTY);
+		if (tal_count(outpoint) < 3 || strlen(outpoint[0]) != 64)
+			break;
+
+		struct bitcoin_txid *txid = tal(tmpctx, struct bitcoin_txid);
+		unsigned int *outnum = tal(tmpctx, unsigned int);
```

Comment:
It's weird to see `unsigned int`s allocated, but maybe I'm just new to this.

You probably want to allocate these off of the same thing as the array that you're putting them in, either `cmd` or `outnums`.

Reply:
I allocated them off `tmpctx` because they are temo variable, however I'm also new to this and am not sure it's the correct way.